util/i2csvg/README.md

```diff
@@ -0,0 +1,159 @@
+
+To assist in parsing debug output the tool can be given a prefix.
+Only lines with the prefix will be processed (after removal of the prefix).
+
+### Textual Source Format
+
+The textual description includes the flags and value written to the I2C host interface fifo to make the transaction.
+The flags and data may be separated by whitespace.
+The flags are letters representing each of the 5 control bits written to the fifo:
+
+|Flag | Use |
+|-----|-----|
+|S | Send a start signal before the data |
+|P | Send a stop signal after the data |
+|R | Read transaction (data value indicates how many bytes to read) |
+|C | With R, allows read to continue by ACK of the last byte read |
+|N | NACK OK, do not raise an error on a NACK |
+
+The value may be an integer (binary, octal, decimal and hex supported) or one of the special values that change how the output is drawn:
+
+|Value | Use |
+|------|-----|
+|A0 | Address with direction bit set to 0 (write) |
+|A1 | Address with direction bit set to 1 (read) |
+|A2 | Address in Quick command with direction bit used for data |
+|M  | Multi-byte transaction, drawn as D1, D2 ... DN (like SMBus) |
+|'tag'| In single quotes: Tag for the value that is drawn in output |
+
+Lines of the input may consist of a single flag+value per line or a comma separated list of flag+value pairs.
+
+### Hexdump source format
+
+In the hexdump source format tool processes the raw 13-bit data that is written to the I2C fifo.
+This is normally in hex (preceded with 0x) but the tool will accept binary, octal or decimal.
+Lines of input may consist of a single hex value, multiple values separated by commas (and optionally whitespace) or multiple values separated by whitespace.
+
+### Output file format
+
+The output file generated depends on the options specified.
+
+|Type | Options|
+|-----|--------|
+|.svg  | If multiple output files are generated or debug/text not selected |
+|.html | If debug/text is selected and svg is generated |
+|.txt  | If debug/test is selected and no svg is generated (--nosvg) |
+
+The example commands assume `$REPO_TOP` is set to the toplevel directory
+of the repo.
+
+### Setup
+
+The i2ctool does not depend on any other packages.
+
+### Examples using i2csvg.py
+
+The i2csvg/examples directory contains some example I2C files.
+For simple examples text can be piped in to the tool using echo.
+
+A simple I2C write consists of a Start with Address and direction bit of zero, and a single data byte (4 in the example) with a Stop.
+
+The baseline tool will generate an svg file (which can be viewed in a browser):
+
+```console
+$ cd $REPO_TOP/util
+$ echo "SA0, P4" | ./i2csvg.py > out.svg
+```
+
+The text version (which is just the input with more structured formatting) can be viewed in the terminal by suppressing the svg output:
+
+```console
+$ cd $REPO_TOP/util
+$ echo "SA0, P4" | ./i2csvg.py --text --nosvg
+S.... A0
```

Comment:
What exactly do the dots mean?  I see a lot of transactions with dots or with out dots.  Is the idea to make an optional 5-bit string to match the 5-flags?

Reply:
On output I pretty-print line up the flags and put dots where they are not set, which of course also aligns the value (assuming fixed-width font). Could change to spaces but dots looked better to me.